xml/cl.xml

```diff
@@ -3462,6 +3469,20 @@ server's OpenCL/api-docs repository.
             <param>const <type>cl_event</type>*                         <name>event_wait_list</name></param>
             <param><type>cl_event</type>*                               <name>event</name></param>
         </command>
+        <command>
+            <proto><type>cl_int</type>                                  <name>clGetLayerInfo</name></proto>
+            <param><type>size_t</type>                                  <name>param_value_size</name></param>
+            <param><type>cl_layer_info</type>                           <name>param_name</name></param>
+            <param><type>void</type>*                                   <name>param_value</name></param>
+            <param><type>size_t</type>*                                 <name>param_value_size_ret</name></param>
+        </command>
+        <command>
+            <proto><type>cl_int</type>                                  <name>clInitLayer</name></proto>
+            <param><type>cl_uint</type>                                 <name>num_entries</name></param>
+            <param>const<type>struct _cl_icd_dispatch</type>*           <name>target_dispatch</name></param>
```

Comment:
It looks like other APIs include a space after `const` by convention, so I'd recommend adding one here also.

This will only matter when we start to generate extension headers, but hopefully that will be soon.

Reply:
Also, this is purely cosmetic, but should we be using the typedef'd name `cl_icd_dispatch` instead of `struct _cl_icd_dispatch`?

Reply:
> It looks like other APIs include a space after `const` by convention, so I'd recommend adding one here also.
> 
> This will only matter when we start to generate extension headers, but hopefully that will be soon.

Fixed.

Reply:
> Also, this is purely cosmetic, but should we be using the typedef'd name `cl_icd_dispatch` instead of `struct _cl_icd_dispatch`?

This I think touches the real issues: `cl_icd.h` is not included by `opencl.h` or extensions headers. `cl_icd_layer.h` pulls `cl_icd.h` to get the definition. I think this prevents including the extension definitions into normal extension headers. I think the header should go along with `cl_icd.h` in the `CL` directory.

The question that is raised now is do we want those files generated alongside the others, especially `cl_icd.h`, and if yes, how much information do we need to add to the xml to be able to generate `cl_icd.h`. Or do we only include the type defined in 'cl_icd.h` in the xml, wich would allow us to use the typedef'd `cl_icd_dispatch`.

What are your views on the problem?

Reply:
`cl_icd.h` is already in the API headers ([link](https://github.com/KhronosGroup/OpenCL-Headers/blob/master/CL/cl_icd.h)), so we could include it from either the extensions header `cl_ext.h` or a new layer-specific header.

I think it would be preferable to put the layer stuff in the same API headers vs. adding a dependency on another repo.  I don't have a strong opinion whether the layer stuff goes into the same extension header or in a layer-specific header, though.

Reply:
To make things clear, I agree with you we should put it in the same directory, and I think, to avoid including 'cl_icd.h' everywhere it should go in its own file.

Reply:
> `cl_icd.h` is already in the API headers ([link](https://github.com/KhronosGroup/OpenCL-Headers/blob/master/CL/cl_icd.h)), so we could include it from either the extensions header `cl_ext.h` or a new layer-specific header.

Indeed, but it is not in the xml file. How should I introduce it? How much of it should be introduced?